safenode/src/network/cmd.rs

```diff
@@ -43,11 +50,58 @@ pub enum SwarmCmd {
         resp: Response,
         channel: ResponseChannel<Response>,
     },
+    RegisterProvidedData {
+        record: Record,
+    },
+    /// Get data from the kademlia store
+    GetProvidedData {
+        key: RecordKey,
+        sender: oneshot::Sender<QueryResponse>,
+    },
 }
 
 impl SwarmDriver {
     pub(crate) fn handle_cmd(&mut self, cmd: SwarmCmd) -> Result<(), Error> {
         match cmd {
+            SwarmCmd::GetProvidedData { key, sender } => {
+                match self.swarm.behaviour_mut().kademlia.store_mut().get(&key) {
+                    Some(data) => {
+                        sender
+                            .send(crate::protocol::messages::QueryResponse::GetChunk(Ok(
+                                Chunk::new(Bytes::from(data.value.clone())),
+                            )))
+                            .map_err(|_| Error::InternalMsgChannelDropped)?;
+                    }
+                    None => {
+                        // empty response here just to test out
+                        sender
+                            .send(crate::protocol::messages::QueryResponse::GetChunk(Err(
```

Comment:
Bit more readable to move the long qualification to use statenent at top of doc.